mage_ai/data_preparation/models/pipeline.py

```diff
@@ -1106,6 +1167,87 @@ async def update(self, data, update_content=False):
                     cache.remove_pipeline(tag_uuid, old_uuid)
                 cache.add_pipeline(tag_uuid, self)
 
+    # Updates pipeline YAML for new pipeline or block names from the import process
+    # compiles list of files to be written
+    # also returns the new pipeline name and the config destination path
+    def __update_pipeline_yaml(self, tmp_dir, config_zip_path, overwrite):
+        files_to_be_written = []
+        with open(config_zip_path, 'r') as pipeline_config:
+            config = yaml.safe_load(pipeline_config)
+
+            # check if pipeline exists with same uuid and generate new one if necessary
+            if not overwrite:
+                uuid = config['uuid']
+                index = 0
+                while self.exists(uuid):
+                    index += 1
+                    uuid = f'{config["uuid"]}_{index}'
+                config['uuid'] = uuid
+                config['name'] = uuid
+            pipeline_name = config['uuid']
+
+            # retain block upstream and downstream references
+            keys_to_keep = ['upstream_blocks', 'downstream_blocks']
+            block_hierarchy = {
+                block['uuid']: {key: [] for key in keys_to_keep}
+                for block in config['blocks']
+            }
+
+            for b_index, block in enumerate(config['blocks']):
+                name = block['name']
+                uuid = block['uuid']
+                type = block['type']
+                language = block.get('language', 'python')
+
+                block_inst = Block(name=name, uuid=uuid, block_type=type, language=language)
+
+                # check if block exists with same uuid and generate new one if necessary
+                if not overwrite:
+                    index = 0
+                    while block_inst.exists:
+                        index += 1
+                        block_inst.uuid = f'{block["uuid"]}_{index}'
+
+                # save block
+                block_destination_path = block_inst.file_path
+                file_extension = block_destination_path.split('.')[-1]
+                block_directory = block_destination_path.split(os.path.sep)[-2]
+                block_zip_path = self.__find_pipeline_file(
+                    self, tmp_dir, f'{uuid}.{file_extension}', block_directory
```

Comment:
remove `self` argument

Reply:
I had issues with calling this method until I realized I was doing it from a static context. Modified both `__find_pipeline_files` and `__update_pipeline_yaml` to class methods.